### PR TITLE
feat: enable facebook oauth in ui-demo

### PR DESCRIPTION
**Pull request by linnall**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the OAuth authentication experience by adding support for multiple providers and updating related components to handle the new configuration.

### Detailed summary
- Updated `<ContinueWithOAuth />` to accept `provider` prop.
- Changed `oauthProvider` type to `KnownAuthProvider`.
- Modified `OAuthConnectionFailed` to use `provider` prop.
- Introduced `oAuthMethods` configuration for Google, Facebook, and Auth0.
- Updated `Authentication` component to manage OAuth states.
- Refactored logic for handling OAuth methods in `ConfigContextProvider`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: examples/ui-demo/src/app/config.tsx [hunk 1]
```diff
@@ -10,8 +10,12 @@ export type Config = {
     showExternalWallets: boolean;
     showPasskey: boolean;
     addPasskey: boolean;
-    showSocial: boolean;
-    addGoogleAuth: boolean;
+    showOAuth: boolean;
+    oAuthMethods: {
+      google: Boolean;
+      facebook: Boolean;
+      auth0: Boolean;
```
Comment (dphilipson) [inline examples/ui-demo/src/app/config.tsx:17]: Should use `boolean` over `Boolean`: there basically isn't ever any reason to use `Boolean`.

**Review by dphilipson — APPROVED** [review]
Looks great! One nitpick.

File: examples/ui-demo/src/app/config.tsx (hunk 1 above)
Comment (linnall) [reply examples/ui-demo/src/app/config.tsx:17]: good catch

**Review by dphilipson — APPROVED** [review]
Looks good!

**Pull request merged**